ray-operator/test/e2e/raycluster_gcs_ft_test.go

```diff
@@ -0,0 +1,238 @@
+package e2e
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appsv1ac "k8s.io/client-go/applyconfigurations/apps/v1"
+	corev1ac "k8s.io/client-go/applyconfigurations/core/v1"
+	metav1ac "k8s.io/client-go/applyconfigurations/meta/v1"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/common"
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/utils"
+	rayv1ac "github.com/ray-project/kuberay/ray-operator/pkg/client/applyconfiguration/ray/v1"
+	. "github.com/ray-project/kuberay/ray-operator/test/support"
+)
+
+const (
+	rayImage   = "rayproject/ray:2.40.0"
+	rayVersion = "2.40.0"
+)
+
+func TestRayClusterGCSFaultTolerence(t *testing.T) {
+	test := With(t)
+	g := NewWithT(t)
+
+	// Create a namespace
+	namespace := test.NewTestNamespace()
+	testScriptAC := newConfigMap(namespace.Name, files(test, "test_detached_actor_1.py", "test_detached_actor_2.py"))
+	testScriptAC = testScriptAC.WithName("test-script")
+	testScriptCM, err := test.Client().Core().CoreV1().ConfigMaps(namespace.Name).Apply(test.Ctx(), testScriptAC, TestApplyOptions)
+	g.Expect(err).NotTo(HaveOccurred())
+	t.Log(testScriptCM.Name)
+	redisCM := corev1ac.ConfigMap("redis-config", namespace.Name).WithLabels(map[string]string{"app": "redis"}).
+		WithData(map[string]string{
+			"redis.conf": `dir /data
+							port 6379
+							bind 0.0.0.0
+							appendonly yes
+							protected-mode no
+							requirepass 5241590000000000
+							pidfile /data/redis-6379.pid`,
+		})
```

Comment:
The indentions make the configuration hard to read, but I think we should have better started the Redis without this configuration map. It is quite irrelevant to the test, but it takes too many lines.

Reply:
The only thing we need is `requirepass 5241590000000000`. You may take this as an example to start a redis without the configmap:
<img width="1014" alt="image" src="https://github.com/user-attachments/assets/1f5670c7-4816-44b6-8a13-51f0146727eb" />